packages/server-wallet/src/protocols/__test__/defund-channel.test.ts

```diff
@@ -0,0 +1,175 @@
+import {defaultTestConfig} from '../..';
+import {createLogger} from '../../logger';
+import {DBDefundChannelObjective, ObjectiveModel} from '../../models/objective';
+import {Store} from '../../wallet/store';
+import {testKnex as knex} from '../../../jest/knex-setup-teardown';
+import {seedAlicesSigningWallet} from '../../db/seeds/1_signing_wallet_seeds';
+import {channel} from '../../models/__test__/fixtures/channel';
+import {Channel} from '../../models/channel';
+import {MockChainService} from '../../chain-service';
+import {WalletResponse} from '../../wallet/wallet-response';
+import {ChannelDefunder} from '../defund-channel';
+import {ChallengeStatus} from '../../models/challenge-status';
+import {stateVars} from '../../wallet/__test__/fixtures/state-vars';
+import {stateWithHashSignedBy} from '../../wallet/__test__/fixtures/states';
+import {alice, bob} from '../../wallet/__test__/fixtures/signing-wallets';
+
+const logger = createLogger(defaultTestConfig());
+const timingMetrics = false;
+
+let store: Store;
+beforeEach(async () => {
+  store = new Store(
+    knex,
+    defaultTestConfig().metricsConfiguration.timingMetrics,
+    defaultTestConfig().skipEvmValidation,
+    '0'
+  );
+
+  await store.dbAdmin().truncateDB();
+  await seedAlicesSigningWallet(knex);
+});
```

Comment:
I imagine that this is a common pattern in the server-wallet codebase now. Is there a test utility to dry up this test setup code?

Reply:
I think there is a benefit to explicitly doing the setup. From this, you know that before each test
- creating a new `store`
- truncating the database
- seeding only alice's signing wallet

Perhaps I'm being paranoid but I'm worried about some utility obscuring some of this.  Explicitly saying who's wallet you'll be using and whether you're wiping the database seems to be pretty relevant info that various by test.

I'd be happy with something like this?:
```
const store = TestUtil.createStore();
await TestUtil.truncate();
await seedAlicesSigningWallet(knex);
```

Reply:
Assuming that it's commonly used (which I think it is), I think the fact that there's already `seedAlicesSigningWallet` is an indication that it's a common enough pattern to encapsulate all that into something like
```
store = TestUtil.blankStore(knex, {seededWalet: 'Alice'})
```

Reply:
I think your suggestion is also fine.

When I also see this [over here](https://github.com/statechannels/statechannels/pull/3146/files#diff-039e86d4e9d5638d7541ebf8d0a2922f71c752b20b6cb6c15165b454ab664899R93-R105), it seems like a code smell. What happens when you add a new required argument to the store function? You'd then need to update every test file that creates a store.